File: Techniques/Two Pointers/SumEqualToPairOfElements.cpp
#include<iostream>
using namespace std;


inline bool isPairSum(int arr[], int n, int x)
{
	int i, j; 
	i = 0; j = n - 1;

	int current_sum = 0;
	while(i < j)
	{
		current_sum = arr[i] + arr[j];
		if(current_sum == x)
			return true;

		if(current_sum < x)
			i++;
		else
			j--;
	}
	return false;
}


int main()
{
	int n; cin >> n;
	int arr[n];

	for(int i=0; i<n; ++i)
		cin >> arr[i];

	int x; cin >> x;

	cout << isPairSum(arr, n, x) << endl;
	return 0;
}

File: Techniques/Two Pointers/SumEqualToPairOfElements.py
#  problem statement in the wiki (Techniques--Two pointers)

def isPairSum(arr, n, x):
	i, j = 0, n - 1

	while i < j:
		current_sum = arr[i] + arr[j]
		if current_sum == x:
			return True

		if current_sum < x:
			i += 1
		else:
			j -= 1
	return False


arr = list(map(int, input().split()))
n = len(arr)
x = int(input())
print(isPairSum(arr, n, x))
